src/App.tsx

```diff
@@ -15,33 +14,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const goods = goodsFromServer;
+
+  const [isVisible, setVisible] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isSorted, setIsSorted] = useState(false);
+  const [sortBy, setSortBy] = useState('alphabetically');
+
+  const visibleGoodsList = () => setVisible(!isVisible);
+
+  const reverse = () => setIsReversed(!isReversed);
+
+  const reset = () => {
+    setSortBy('');
+    setIsReversed(false);
+    setIsSorted(false);
+  };
+
+  const sortByLength = () => {
+    setIsSorted(true);
+    setSortBy('length');
+  };
+
+  const sortByAlphabetically = () => {
+    setIsSorted(true);
+    setSortBy('alphabet');
+  };
+
+  const newGoods = [...goods];
+
+  if (isSorted) {
+    newGoods.sort((good1, good2): number => {
+      switch (sortBy) {
+        case 'length':
+          return good1.length - good2.length;
+
+        case 'alphabet':
+          // I added "if" because if the array is flipped it will not sort correctly
+          if (!isReversed) {
+            return good1.localeCompare(good2);
+          }
+
+          return good2.localeCompare(good1);
+
+        default: return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    newGoods.reverse();
+  }
+
+  return (
+    <div className="App">
+      {!isVisible && (
+        <button
+          type="button"
+          className="App__button"
+          onClick={visibleGoodsList}
+        >
+          Start
+        </button>
+      )}
+
+      {isVisible && (
+        <div className="App__container">
+          <ul className="App__list">
+            {(newGoods.map((good) => (
+              <li
+                key={good}
+                className="App__item"
+              >
+                {good}
+              </li>
+            )))}
+          </ul>
+
+          <div className="App__serviceContainer-button">
```

Comment:
What is “service”? I can't understand. It also does not look like consistent naming. 
If you need several words – you can write something like this `App__service-container-button`